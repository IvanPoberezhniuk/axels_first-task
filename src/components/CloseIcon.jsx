import React from 'react';
import PropTypes from 'prop-types';
import {CloseIconStyles} from '../styled/components/CloseIcon';

const CloseIcon = ({onIconClick}) => {
  return (
    <CloseIconStyles onClick={onIconClick}/>
  );
};

CloseIcon.propTypes = {
  onIconClick: PropTypes.func
};

export default CloseIcon;
