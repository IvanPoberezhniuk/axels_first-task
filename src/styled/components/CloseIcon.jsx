import styled from 'styled-components';


const CloseIconStyles = styled.div`
  position: absolute;
  right: 1rem;
  top: 1rem;
  width: 32px;
  height: 32px;
  z-index: 22;

  :hover {
    opacity: 1;
    cursor: pointer;
  }

  :before, :after {
    position: absolute;
    left: 15px;
    content: ' ';
    height: 33px;
    width: 2px;
    background-color: #333;
  }

  :before {
    transform: rotate(45deg);
  }

  :after {
    transform: rotate(-45deg);
  }
`;


export {
  CloseIconStyles
};
