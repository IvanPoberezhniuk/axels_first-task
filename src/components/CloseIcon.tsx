import React from 'react';

import { CloseIconStyles } from '../styled/components/CloseIcon';

interface Props {
  onIconClick: () => void;
}

const CloseIcon = ({ onIconClick }: Props) => (
  <CloseIconStyles onClick={onIconClick} />
);

export default CloseIcon;
