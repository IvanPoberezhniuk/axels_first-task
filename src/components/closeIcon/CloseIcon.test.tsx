import { shallow, ShallowWrapper } from 'enzyme';

import { CloseIcon } from '../index';

import { Props } from './CloseIcon';

let component: ShallowWrapper;
const setUp = (props: Props) => shallow(<CloseIcon {...props} />);

beforeEach(() => {
  component = setUp({ onIconClick: jest.fn() });
});

describe('CloseIcon component', () => {
  it('should take snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});

