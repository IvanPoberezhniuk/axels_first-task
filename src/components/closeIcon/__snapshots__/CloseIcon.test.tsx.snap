// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseIcon component SNAPSHOT 1`] = `
<styled.div
  onClick={[MockFunction]}
/>
`;
